topics/admin/tutorials/jenkins/tutorial.md

```diff
@@ -0,0 +1,490 @@
+>
+>    2. Click "Add user or group" and add the `admin` user
+>
+>    3. Configure the permissions like below. Clicking on the checked checkbox icon at the right will allow you to enable/disable all of the checkboxes for that row.
+>
+>       ![Configuring matrix based permissions](../../images/jenkins-09-permissions.png)
+>
+>       The matrix based security plugin is provided by a plugin, we installed this by setting the variable `jenkins_plugins: [matrix-auth]` in our group variables earlier.
+>       The checkboxes give you the ability to grant or deny permissions to certain classes of users. Each column is an individual permission, and you can hover over each column to get more information about what that permission allows.
+>
+>       In the above example we have configured that:
+>         - *Anonymous users* (not logged in) may `Read`, so they can access the Jenkins page, but by default will not see any jobs. The `Discover` permission is also granted so that if a user has bookmarked the ansible-galaxy job page, they will automatically be redirected through the login page, if they are not already logged in.
+>         - Once they login and are *Authenticated Users*, they can see that `Manage` and `Run` Jobs and Views
+>         - Only the *admin* user, however, can configure `Agents` to run those jobs.
+>
+>    3. Under **CSRF Protection**, check the box to prevent CSRF, and select the default crumb issuer, and then enable proxy compatibility.
+>
+> 4. Save
+>
+>    > ### {% icon question %} Question
+>    >
+>    > Try accessing your Jenkins from a private browsing session. How does it look?
+>    >
+>    > > ### {% icon solution %} Solution
+>    > > It should appear as if there are no jobs, until you log in.
+>    > {: .solution}
+>    {: .question}
+>
+{: .hands_on}
+
+# Advanced Topics
+
+These topics are outside the scope of a training or do not lend themselves well to interesting exercises, but some discussion and configuration information will be provided below documenting common issues and questions that can be useful if you start using Jenkins "in anger".
+
+## Updating Jenkins
+
+Jenkins updates regularly due to security issues. We recommend that you either:
+
+1. Regularly check the [Jenkins Changelog](https://jenkins.io/changelog/), and update the server whenever there are important security fixes
+2. OR include some sort of yum or apt auto-updater on whichever system that Jenkins is running on
+
+The downside of automatic updating is that Jenkins will restart and kill any jobs that had been running. Choose which works best for your environment. UseGalaxy.eu chose a completely automated setup because their jobs can be killed and few things are affected. UseGalaxy.org chose a more manual setup because their jobs are more critical, there if jobs fail, Galaxy pull request tests will fail unexpectedly and unhelpfully.
```

Comment:
I think jenkins will only restart when no job is running, so it waits for running jobs, right? At least I have seen this happening.

Reply:
Ok, was mentioned a few sentences below :)